File: typescript/src/entity.ts
export interface Entity {
    name: string,
    hp: number,
}

export function newEntity(name: string): Entity {
    return {
        name: name,
        hp: 100
    }
}

export function dealDamage(dealer: Entity, target: Entity, damage: number) {
    target.hp -= damage;
    console.log(`${dealer.name} Damage <>> ${target.name} with ${damage} dmg\n`);
    console.log("===============================\n");
}

export function assignAttackDamage(max: number) {
    return Math.ceil(Math.random() * max);
}

File: cpp/src/Entity.cpp
#include <iostream>
#include "Entity.h"

std::string Entity::get_name() 
{
	return name;
}

Entity::Entity() 
{
	this->name = "NoName Entity";
	this->hp = 100;
}

Entity::Entity(const char *name)
{
	this->name = name;
	this->hp = 100;
}

int Entity::get_hp() 
{
	return hp;
}

void Entity::set_hp(int value) 
{
	hp = value;
}

int Entity::assign_attack_damage(int max) 
{
	return rand() % max + 1;
}
void Entity::deal_damage_to(Entity *entity, int damage) 
{
	int entity_hp = entity->get_hp();
	entity->set_hp(entity_hp - damage);
	std::cout << name << " Damage: " << entity->get_name() << " with " << damage << " dmg\n";
	std::cout << "===============================\n";
}